### DISCUSSION TITLE
BombSquad.exe from 1.6.5 WIndows build identified as malicious

### DISCUSSION THREAD

**Discussion by 7wells** [efroemling/ballistica#349] [General]
@efroemling 

Hi Eric, I hope my message finds you well! :smiley:

Various virus scanners identify the executable `BombSquad.exe` in your official build 1.6.5 from [there](https://files.ballistica.net/bombsquad/builds/BombSquad_Windows_1.6.5.zip) as malicious:

https://www.virustotal.com/gui/file/adb8ac26349eda58326a8c6c206443be0c543fcc1ab14ed6d65a8d2d5d1b02e9

![grafik](https://user-images.githubusercontent.com/65889763/147228463-689fd659-6325-45b3-a91d-2a6c0377f98e.png)

This was a confirmation of my initial finding with IObit Malware Fighter 9.

Is this a false positive? And/or related to a problem with Visual Studio? (see hints [there](https://developercommunity.visualstudio.com/search?space=8&q=Fugrafa))

Would you mind checking if someone hacked your site and manipulated your files? Furthermore, could you please consider adding checksums (MD5, SHA, whatever) in a safe place (e.g., here on GitHub in addition to the download site?) for verification that the files have not been tampered with? Thank you so much!

I wish you and everybody peaceful Christmas Holidays, and most of all, good health and all the best for 2022! :four_leaf_clover:

**Comment by efroemling** [comment]
Thanks for the heads-up.
My assumption is that this was a false positive.
I did go ahead and add a checksum stage to my build scripts. My build server will now write out checksums to a separate server just before it uploads the packages to files.ballistica.net.
You can access the current checksums at https://tools.ballistica.net/checksums

I would be curious if the latest set of builds still triggers those same false positives.

I'll go ahead and add an 'official' link to those builds from the ballistica GitHub README and include the checksum link there too.

**Comment by efroemling** [comment]
Quick further followup on this: instead of the separate checksums page, I went ahead and put together an automated builds page at https://tools.ballistica.net/builds which contains these checksums as well as direct links to the corresponding files and extra info such as build-numbers. A link is available in the Resources section of the new tools portal at https://tools.ballistica.net.
Checksum data still lives on a completely different server than the files themselves so hopefully any sort of tampering would be quite a tricky task.
